### PR TITLE
Add matter integration

### PR DESCRIPTION
**Pull request by MartinHjelmare**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/83064
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/3766
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
We need to add some more information about the requirements for the MVP docs.

File: source/_integrations/matter.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ha_config_flow: true
+ha_codeowners:
+  - '@MartinHjelmare'
+ha_domain: matter
+ha_platforms:
+  - light
+ha_integration_type: integration
+---
+
+This integration allows you to control a Matter network.
```
Comment (marcelveldt) [inline source/_integrations/matter.md]: I'd rephrase this a bit so it will be about controlling Matter devices and not a Matter network.

Comment (MartinHjelmare) [reply source/_integrations/matter.md]:
```suggestion
This integration allows you to communicate with and control Matter devices.
```

**Comment by MartinHjelmare** [comment]
Base integration has been merged to a separate target branch in core repo. It's not on dev yet.

File: source/_integrations/matter.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+title: Matter
+description: Instructions on how to integrate Matter with Home Assistant.
+featured: true
+ha_category:
+  - Light
+ha_release: '2022.11'
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
ha_release: '2022.12'
```

File: source/_integrations/matter.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+title: Matter
+description: Instructions on how to integrate Matter with Home Assistant.
+featured: true
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
```

File: source/_integrations/matter.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+title: Matter
+description: Instructions on how to integrate Matter with Home Assistant.
+ha_category:
+  - Light
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md:6]:
```suggestion
  - Binary Sensor
  - Light
  - Sensor
  - Switch
```

File: source/_integrations/matter.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ha_release: '2022.12'
+ha_iot_class: Local Push
+ha_config_flow: true
+ha_codeowners:
+  - '@MartinHjelmare'
+ha_domain: matter
+ha_platforms:
+  - light
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md:18]:
```suggestion
  - binary_sensor
  - light
  - sensor
  - switch
```

File: source/_integrations/matter.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+title: Matter
+description: Instructions on how to integrate Matter with Home Assistant.
+ha_category:
+  - Light
+ha_release: '2022.12'
+ha_iot_class: Local Push
+ha_config_flow: true
+ha_codeowners:
+  - '@MartinHjelmare'
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md:14]:
```suggestion
  - '@marcelveldt'
  - '@MartinHjelmare'
```

File: source/_integrations/matter.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ha_domain: matter
+ha_platforms:
+  - binary_sensor
+  - light
+  - sensor
+  - switch
+ha_integration_type: integration
+---
+
+This integration allows you to communicate with and control Matter devices.
```
Comment (marcelveldt) [inline source/_integrations/matter.md]: Do we want to mention the dependency on the Matter server add-on ?
Basically Matter is now only available for HAOS users.

Comment (MartinHjelmare) [reply source/_integrations/matter.md]: I think we should mention the matter server dependency. If you can't run the add-on you can run the server some other way, if you know how. Ie, I think we can add a section for advanced installation instructions similar like we have for Z-Wave JS.

Comment (marcelveldt) [reply source/_integrations/matter.md]: OK, hold on, I'll write something down based on zwave_js

File: source/_integrations/matter.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+{% include integrations/config_flow.md %}
```
Comment (marcelveldt) [inline source/_integrations/matter.md:30]:
```suggestion
{% include integrations/config_flow.md %}

If you run Home Assistant Container, Home Assistant Core, or you don’t want to use the built-in Matter Server add-on, please see the [advanced installation instructions](https://www.home-assistant.io/integrations/matter/#advanced-installation-instructions).


## Advanced installation instructions

If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Matter Server add-on, you will need to run the Matter server yourself, to which the Matter integration will connect.

### Running [Matter Server](https://github.com/home-assistant-libs/python-matter-server).

This application provides the connection between your Matter network (called Fabric in technical terms) and Home Assistant. The Home Assistant Matter integration connects to this server via a websocket connection. You need to run this Matter server before you can use the integration.

There are multiple ways to run this server:

**Option 1: The official MatterServer add-on, as described above**

_This option is only available for Home Assistant OS (the recommended installation type) and Home Assistant Supervised installations._

**Option 2: Run the Matter server yourself**

This option is considered a very advanced setup and only for experienced users. You can find instructions on how to run the Matter server here: https://github.com/home-assistant-libs/python-matter-server

```

File: source/_integrations/matter.md (hunk 7 above)
Comment (marcelveldt) [inline source/_integrations/matter.md]:
```suggestion
This integration allows you to communicate with and control Matter devices on your local WiFi or Thread network.

<p class='note'>
Both Matter and the implementation within Home Assistant are in early (v1) state and probably features are missing or could be improved.
</p>
```

File: source/_integrations/matter.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  - light
+  - sensor
+  - switch
+ha_integration_type: integration
+---
+
+This integration allows you to communicate with and control Matter devices on your local WiFi or Thread network.
+
+<p class='note'>
+Both Matter and the implementation within Home Assistant are in early (v1) state and probably features are missing or could be improved.
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
Both Matter and the implementation within Home Assistant are in early (v1) state and features may be missing or could be improved.
```

File: source/_integrations/matter.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+</p>
+
+{% include integrations/config_flow.md %}
+
+If you run Home Assistant Container, Home Assistant Core, or you don’t want to use the built-in Matter Server add-on, please see the [advanced installation instructions](https://www.home-assistant.io/integrations/matter/#advanced-installation-instructions).
+
+
+## Advanced installation instructions
+
+If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Matter Server add-on, you will need to run the Matter server yourself, to which the Matter integration will connect.
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Matter Server add-on, you will need to run the Matter Server yourself, to which the Matter integration will connect.
```

File: source/_integrations/matter.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+### Running [Matter Server](https://github.com/home-assistant-libs/python-matter-server).
+
+This application provides the connection between your Matter network (called Fabric in technical terms) and Home Assistant. The Home Assistant Matter integration connects to this server via a websocket connection. You need to run this Matter server before you can use the integration.
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
This application provides the connection between your Matter network (called Fabric in technical terms) and Home Assistant. The Home Assistant Matter integration connects to this server via a websocket connection. You need to run the Matter Server before you can use the integration.
```

File: source/_integrations/matter.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+There are multiple ways to run this server:
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
There are multiple ways to run the server:
```

File: source/_integrations/matter.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+
+**Option 1: The official MatterServer add-on, as described above**
+
+_This option is only available for Home Assistant OS (the recommended installation type) and Home Assistant Supervised installations._
+
+**Option 2: Run the Matter server yourself**
+
+This option is considered a very advanced setup and only for experienced users. You can find instructions on how to run the Matter server here: https://github.com/home-assistant-libs/python-matter-server
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
This option is considered a very advanced setup and only for experienced users. You can find instructions on how to run the Matter Server here: https://github.com/home-assistant-libs/python-matter-server
```

**Review by marcelveldt — DISMISSED** [review]

File: source/_integrations/matter.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+This integration allows you to communicate with and control Matter devices on your local WiFi or Thread network.
+
+<p class='note'>
+Both Matter and the implementation within Home Assistant are in early (v1) state and probably features are missing or could be improved.
+</p>
+
+{% include integrations/config_flow.md %}
+
+If you run Home Assistant Container, Home Assistant Core, or you don’t want to use the built-in Matter Server add-on, please see the [advanced installation instructions](https://www.home-assistant.io/integrations/matter/#advanced-installation-instructions).
```
Comment (frenck) [inline source/_integrations/matter.md]:
```suggestion
If you run Home Assistant Container, Home Assistant Core, or you don’t want to use the built-in Matter Server add-on, please see the [advanced installation instructions](#advanced-installation-instructions).
```

File: source/_integrations/matter.md (hunk 13 above)
Comment (frenck) [inline source/_integrations/matter.md]: Avoid bare URLs, link the text instead.

Comment (MartinHjelmare) [reply source/_integrations/matter.md]:
```suggestion
This option is considered a very advanced setup and only for experienced users. You can find instructions on how to run the Matter Server in the [project repository](https://github.com/home-assistant-libs/python-matter-server).
```

File: source/_integrations/matter.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+{% include integrations/config_flow.md %}
+
+If you run Home Assistant Container, Home Assistant Core, or you don’t want to use the built-in Matter Server add-on, please see the [advanced installation instructions](#advanced-installation-instructions).
+
+
+## Advanced installation instructions
+
+If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Matter Server add-on, you will need to run the Matter Server yourself, to which the Matter integration will connect.
+
+### Running [Matter Server](https://github.com/home-assistant-libs/python-matter-server).
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
### Running [Matter Server](https://github.com/home-assistant-libs/python-matter-server)
```

File: source/_integrations/matter.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+{% include integrations/config_flow.md %}
+
+If you run Home Assistant Container, Home Assistant Core, or you don’t want to use the built-in Matter Server add-on, please see the [advanced installation instructions](#advanced-installation-instructions).
+
+
+## Advanced installation instructions
+
+If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Matter Server add-on, you will need to run the Matter Server yourself, to which the Matter integration will connect.
+
+### Running [Matter Server](https://github.com/home-assistant-libs/python-matter-server)
```
Comment (MartinHjelmare) [inline source/_integrations/matter.md]:
```suggestion
### Running Matter Server
```

Comment (MartinHjelmare) [reply source/_integrations/matter.md]: The link interferes with the anchor, it seems.

**Comment by MartinHjelmare** [comment]
I'm happy with this for now.

File: source/_integrations/matter.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Matter Server add-on, you will need to run the Matter Server yourself, to which the Matter integration will connect.
+
+### Running Matter Server
+
+This application provides the connection between your Matter network (called Fabric in technical terms) and Home Assistant. The Home Assistant Matter integration connects to this server via a websocket connection. You need to run the Matter Server before you can use the integration.
+
+There are multiple ways to run the server:
+
+**Option 1: The official MatterServer add-on, as described above**
```
Comment (marcelveldt) [inline source/_integrations/matter.md]:
```suggestion
**Option 1: The official Matter Server add-on, as described above**
```

**Review by frenck — APPROVED** [review]
